TBRANDS-29 - Hero Block

matthewroach opened the pull request:
## Description

Hero block based on desktop designs

### This has a corresponding PR here:  https://github.com/WPMedia/arc-themes-feature-pack/pull/211

## Designs

https://www.figma.com/file/giQllWgtzGk42KwK2XD9rY/Brand-Browsing-Page-Designs?node-id=246%3A7662

## Jira Ticket

- [TBRANDS-29](https://arcpublishing.atlassian.net/browse/TBRANDS-29)

## Test Steps

### Easy way
1. Checkout this branch `git checkout TBRANDS-29-hero-block`
2. Run storybook: `npm run storybook`
3. Check the `Hero` block stories under the commerce theme.

Note:  See the file `.storybook/themes/commerce.scss` This is inline with what I have set up in the themepack pr:  https://github.com/WPMedia/arc-themes-feature-pack/pull/211

### Extra Credit Way

Requires Feature Pack PR - https://github.com/WPMedia/arc-themes-feature-pack/pull/211

1. Checkout this branch `git checkout TBRANDS-29-hero-block`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/hero-block`
5. Load PageBuilder - Hero Block is now available

Note: You may want to clear you local DB and cache to prevent annoying content errors in pagebuilder, but its not mandatory to complete the testing.

There are two main configurations of the hero block as you have seen from the images above.  Now that you are up and running, let's build first the onlay configuration and then we can easily convert it to the stacked configuration.  So, grab a cup of freshly brewed coffee and let do this!

1. Once you have Pagebuilder running, create a new page called `Hero`.
2. In your new `Hero` page configuration, Choose the `Single Column Regular Width Layout`
4. Add the `Hero` block to the `main` section of the layout.
6. Next let's set up the `Hero` properties.
7. For `Layout configuration`, set it to `overlay`
8. For content alignment, set to `center`
9. For content style, set to `light`
10. For `Image URL for Desktop`  enter: `https://cloudfront-us-east-1.images.arcpublishing.com/corecomponents/443IPW3JPREMTAUXAGC5S24TZA.jpg`
11. For `Image Breakpoint`, select the 800px option **(Would love not to have to do this)**
12. 8. For `Image URL for Mobile`  enter: `https://images.unsplash.com/photo-1515445849-1e4c80a3eed1?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8bWFuJTIwaW4lMjBmb3Jlc3R8ZW58MHx8MHx8&w=1000&q=80`
13. For `headline`, enter: `All The Fall Things`
14. For `subheadline`, enter `Versatile. Slightly-cooler temps and light layers.`
15. For `description`, enter `Versatile. Slightly-cooler temps and light layers.`
16. For button 1: href => `google.com`, Test => `Men's`, Type => Secondary
17. For button 2: href => `google.com`, Test => `Woman's`, Type => Secondary

OK, good job, you should have this displaying in your preview:

![image](https://user-images.githubusercontent.com/833461/160861960-f3608c06-07d9-4f05-ac62-3a14ced9a70e.png)

Now for extra credit, let's change the setting we already have so that it becomes stacked

1. Change the Layout configuration to `stacked`
2. Change Content Style to `dark`
3. For both buttons, change the type to `primary`

Well done!  Now you should have this in your preview:

![image](https://user-images.githubusercontent.com/833461/160862643-dcd3e433-497b-4e74-833c-7b6d0cdf71b0.png)


## Dependencies or Side Effects

Requires Feature Pack PR - https://github.com/WPMedia/arc-themes-feature-pack/pull/211

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

JackHowa committed:
refactor: Prevent direct state management which could cause issues

Fixes https://lgtm.com/projects/g/WPMedia/arc-themes-blocks/?mode=tree&ruleFocus=1823611029

JackHowa committed:
feat: Add show and hide caption for video

JackHowa committed:
feat: Take in credits from article body video

JackHowa committed:
feat: Target all image-metadata including gallery and the video player caption

JackHowa committed:
fix: Remove inaccessible contrast-level color in favor of working, accessible image metadata color

JackHowa committed:
Merge branch 'arc-themes-release-version-1.20' into REFACTOR-change-direct-state-mutation

JackHowa committed:
style: Undo spacing change

JackHowa committed:
feat: Take in item fields based off of ans data

JackHowa committed:
test: Improve coverage and is-serverside

JackHowa committed:
feat: Take in hide video title for lead art block

JackHowa committed:
docs: Add title and caption for lead art video

JackHowa committed:
fix: Add fallbacks as if there's no caption there will be no description object even

JackHowa committed:
feat: Take in fetched as well as hard-coded data from video player

JackHowa committed:
feat: Add caption logic for fetching as well with videos

JackHowa committed:
style: Remove unnecessary console log

JackHowa committed:
fix: Follow ac of taking in proptypes and reusing the hide title for all media items

JackHowa committed:
style: Target image metadata class directly without p tag

JackHowa committed:
fix: Reorder the proptypes so that the spread is below the group so that the proptypes work with arc

Commit:
Merge pull request #1286 from WPMedia/TMEDIA-601-show-caption-video-options

Tmedia 601 show caption video options

Commit:
Merge branch 'arc-themes-release-version-1.20' into REFACTOR-change-direct-state-mutation

Commit:
Merge pull request #1284 from WPMedia/REFACTOR-change-direct-state-mutation

Small Improvement:  Prevent direct state management which could cause issues

matthewroach committed:
Work in progress hero block

Skeleton hero block based on desktop designs

JackHowa commented on blocks/hero-block/_index.scss:
```diff
@@ -0,0 +1,40 @@
+	grid-template: "hero";
+	justify-content: center;
+
+	@include scss.block-properties("hero");
```
should we disable this in stylelint if we're going to go against its recs?

JackHowa replied:
should we disable this in stylelint if we're going to go against its recs?

JackHowa commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,125 @@
+		link2Text,
+		link2Type,
+		subHeadline,
+		variant = "dark",
```
I like this description of `variant`

JackHowa replied:
I like this description of `variant`

JackHowa commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,125 @@
+
+	const classes = [
+		BLOCK_CLASS_NAME,
+		variant === "dark" ? `${BLOCK_CLASS_NAME}--dark` : `${BLOCK_CLASS_NAME}--light`,
```
nice

JackHowa replied:
nice

JackHowa commented on blocks/hero-block/features/hero/default.test.jsx:
```diff
@@ -0,0 +1,32 @@
+	it("should not render headline", () => {
+		render(<Hero customFields={{ imageURL: "image" }} />);
+		expect(screen.getByRole("img")).toBeInTheDocument();
+		expect(screen.queryByText("My Headline")).not.toBeInTheDocument();
```
nice

JackHowa replied:
nice

JackHowa commented on blocks/hero-block/package.json:
```diff
@@ -0,0 +1,26 @@
+  },
+  "peerDependencies": {
+    "@arc-fusion/prop-types": "^0.1.5",
+    "@wpmedia/arc-themes-components": "*"
```
looks good

JackHowa replied:
looks good

JackHowa reviewed (commented):
Looks good to me. And good work live-coding and debugging while doing this

JackHowa committed:
fix: Use subtitle rather than duplicating title

JackHowa committed:
feat: Add static for lead art handling of non-lightbox images

matthewroach committed:
Using latest SDK errors out with VideoPlayer when using blank strings

JackHowa committed:
test: Improve test coverage of somewhat related service to offer. Trigger coverage check

matthewroach committed:
Merge pull request #1299 from WPMedia/TMEDIA-601-video-captions-null-fix

TMEDIA-601 - Blank Strings fix

JackHowa committed:
test: Remove imported component from test. Replace with more precise class check

JackHowa committed:
test: Remove test change

JackHowa committed:
test: Improve coverage with mocking url search params

Commit:
Merge branch 'arc-themes-release-version-1.20' into TMEDIA-708-sub-headline-subs-1.2

Commit:
Merge branch 'arc-themes-release-version-1.20' into TMEDIA-701-fusion-static-lead-art

JackHowa committed:
Revert selectively "feat: Add static for lead art handling of non-lightbox images"

This reverts commit 5a988868ceeb88be9c89826d2b52ca6db42cafba.

Commit:
Merge pull request #1297 from WPMedia/TMEDIA-708-sub-headline-subs-1.2

TMEDIA 708: Use subs subtitle rather than duplicating title

JackHowa committed:
refactor: Remove unreachable code after discussing and investigating.

via https://github.com/WPMedia/arc-themes-blocks/pull/1298#issuecomment-1075210237

Commit:
Merge branch 'arc-themes-release-version-1.20' into TMEDIA-701-fusion-static-lead-art

Commit:
Merge pull request #1298 from WPMedia/TMEDIA-701-fusion-static-lead-art

Tmedia 701 raw html static

matthewroach committed:
Add missing newTab prop for large and medium PromoImage

matthewroach committed:
Merge pull request #1300 from WPMedia/TMEDIA-765-promos-new-tab

TMEDIA-765 - Open in new tab for Large and Medium Promos

Commit:
additioanl hero modifications

Commit:
Merge branch 'arc-themes-release-version-1.20' into TBRANDS-29-hero-block

Commit:
Updates from arc-themes-release-version-1.20

Commit:
Merge with arc-themes-release-version-2.00

matthewroach commented on blocks/hero-block/_index.scss:
```diff
@@ -0,0 +1,79 @@
+		@include scss.block-components("hero-text");
+	}
+
+	&--text.stacked {
```
The `.stacked` selector should be scoped to the block `.b-hero--stacked` otherwise we will have issues if some other block uses a CSS class `stacked`

brentmiller1973 replied:
Added the right conventions inline with your request

JackHowa commented on blocks/hero-block/intl.json:
```diff
@@ -0,0 +1,13 @@
+{
+	"hero-block.hello-text": {
+		"de": "Hallo",
+		"en": "Hello World",
+		"es": "Hola Mundo",
+		"fr": "Bonjour le monde",
+		"ja": "こんにちは世界",
+		"ko": "안녕하세요 세계",
+		"no": "Hei Verden",
+		"pt": "Olá Mundo",
+		"sv": "Hej Verden"
```
these are test texts. if you don't need these, then they can be removed

brentmiller1973 replied:
this has been removed

JackHowa reviewed (commented):
Looks like the package-lock was deleted? if you want the branch's package-lock, you can run `git checkout arc-themes-release-version-2.00 -- package-lock.json`

JackHowa commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,133 @@
+					) : null}
+					{description ? <Paragraph className="b-hero--paragraph">{description}</Paragraph> : null}
+
+					{(link1Action && link1Text) || (link2Action && link2Text) ? (
+						<Stack
+							className="b-hero--buttonrow"
+							direction="horizontal"
+							justification={alignment === "center" ? "center" : "start"}
+						>
+							<Button href={link1Action} variant={link1Type}>
+								{link1Text}
+							</Button>
+							<Button href={link2Action} variant={link2Type}>
+								{link2Text}
+							</Button>
+						</Stack>
+					) : null}
```
not sure if i follow the logic here. I would expect to render each link button only if there's that specific link. For example, if there's link1action and link1text truthy, then link2action and link2text would still be rendered here. can you check if the buttons exist individually as well? 

The top-level logic is ok for rendering the stack container.

JackHowa replied:
Let me know if you need any clarification here @visionbegin. This appears to still be in the code

brentmiller1973 replied:
I got you.

brentmiller1973 replied:
Added the internal check logic

brentmiller1973 replied:
Also see the one button story.

JackHowa commented on blocks/hero-block/index.story.jsx:
```diff
@@ -0,0 +1,15 @@
+		chromatic: { viewports: [320, 1200] },
+	},
+};
+
+export const showHeading = () => <Hero customFields={{ showHeading: true }} />;
+
+export const hideHeading = () => <Hero customFields={{ showHeading: false }} />;
```
I know this is a wip block. but it's a good idea to think about storybook as well here? Personally I start with storybook then use live fusion. but everyone's process is different

brentmiller1973 replied:
Check it out now

Commit:
adding second image

Commit:
Removing unneeded classes due to sass fix

Commit:
Removing unneeded classes due to sass fix

Commit:
Adding tests

Commit:
Adding tests

Commit:
Adding tests

JackHowa commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,158 @@
+					) : null}
+					{description ? <Paragraph>{description}</Paragraph> : null}
+
+					{(link1Action && link1Text) || (link2Action && link2Text) ? (
```
previous comments haven't been addressed yet https://github.com/WPMedia/arc-themes-blocks/pull/1292#discussion_r838795825

Is this ready for review?

brentmiller1973 replied:
This is ready for review.....not sure which comments your are referring to

JackHowa replied:
the comments I linked to  https://github.com/WPMedia/arc-themes-blocks/pull/1292#discussion_r838795825
<img width="869" alt="Screen Shot 2022-04-01 at 13 47 22" src="https://user-images.githubusercontent.com/5950956/161323752-f965a1b1-6782-48a3-b615-cf951194381d.png">

brentmiller1973 replied:
Sorry for confusion.  Those issues have been addressed.

JackHowa reviewed (changes requested):
Not sure if this is ready for review? I noticed one comment was not addressed https://github.com/WPMedia/arc-themes-blocks/pull/1292#discussion_r838795825 https://github.com/WPMedia/arc-themes-blocks/pull/1292#discussion_r838790499

Commit:
Storybook and tests

Commit:
Updated button logic

Commit:
syncing with release 20

matthewroach commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,162 @@
+			</picture>
+
+			<Stack
+				alignment={alignment}
```
The value being passed to `alignment` is giving react prop types errors - The value `left` is being passed which is not a valid value for this property of stack

brentmiller1973 replied:
Alignment has been removed to let it default.

matthewroach commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,162 @@
+							direction="horizontal"
+							justification={alignment === "center" ? "center" : "start"}
+						>
+							{link1Action && link1Text && (
```
We should be using ternary to conditional render 
```
{link1Action && link1Text ? (
	<Button href={link1Action} variant={link1Type}>
		{link1Text}
	</Button>
) : null}
```

brentmiller1973 replied:
ternaries it is

matthewroach commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,162 @@
+									{link1Text}
+								</Button>
+							)}
+							{link2Action && link2Text && (
```
Same as comment above, we should be using ternary

brentmiller1973 replied:
ternaries it is

matthewroach commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,162 @@
+
+			<Stack
+				alignment={alignment}
+				className={`${BLOCK_CLASS_NAME}--text ${BLOCK_CLASS_NAME}--${layout} ${
```
`${BLOCK_CLASS_NAME}--text ${BLOCK_CLASS_NAME}--${layout}`

Do we need both classes here to always be on this element - we could condense it to just use the `${layout}` one and apply the need CSS to the relevant layout class instance

brentmiller1973 replied:
Updated the class names to reflect the layout selected

matthewroach commented on blocks/hero-block/intl.json:
```diff
@@ -0,0 +1,13 @@
+{
```
We can delete this file

matthewroach commented on blocks/hero-block/package.json:
```diff
@@ -0,0 +1,26 @@
+  "files": [
+    "_index.scss",
+    "features",
+    "intl.json"
```
Can remove the `intl.json` line

brentmiller1973 replied:
removed

Commit:
Removed intl file

Commit:
Removed intl file

Commit:
Ok, ternaries it is

Commit:
removed alignment from parent stack

JackHowa commented:
@visionbegin looks like a lot of changes in the package-lock due to the merge. I had a similar comment in Eric's pr with steps to fix as well 

```
git checkout arc-themes-release-version-2.00 -- package-lock.json
npm i 
```
then commit and push your changes


https://github.com/WPMedia/arc-themes-blocks/pull/1302#issuecomment-1084781179

Commit:
updating classes

Commit:
package lock file updated

visionbegin commented:
> @visionbegin looks like a lot of changes in the package-lock due to a bad merge. I had a similar comment in Eric's pr with steps to fix as well
> 
> ```
> git checkout arc-themes-release-version-2.00 -- package-lock.json
> npm i 
> ```
> 
> then commit and push your changes
> 
> [#1302 (comment)](https://github.com/WPMedia/arc-themes-blocks/pull/1302#issuecomment-1084781179)

Thanks @JackHowa .  Its been updated

Commit:
updated stylelint file

Commit:
Retesting stylelint

Commit:
reordering hero styles to bottom of definition

Commit:
Updated spcaing rules

Commit:
Updated style rules

matthewroach committed:
Use BEM naming, add logic for nest heading rendering

matthewroach committed:
Update commerce theme

Commit:
further refinement

nschubach commented on blocks/hero-block/README.md:
```diff
@@ -0,0 +1,44 @@
+## Props
+
+| **Prop**          | **Required** | **Type** | **Descripton** |
+| ----------------- | ------------ | -------- | -------------- |
+| **required prop** | yes          |          |                |
+| **optional prop** | no           |          |                |
+
+## ANS Schema
+
+Outline any schema information requirements necessary to know for ths block
+
+### ANS Fields
+
+- `Add all ANS fields used in the block`
+
+## Internationalization fields
+
+| Phrase key | Default (English)     |
+| ---------- | --------------------- |
+| `key`      | `english translation` |
+
+## Events
+
+Blocks can emit events. The following is a list of events that are emitted by this block.
+
+| **Event Name** | **Description**    |
+| -------------- | ------------------ |
+| **eventName**  | Describe the event |
+
+### Event Listening
+
+Include block specific intructions for event listening.
+
+OR
+
+This block does not emit any events.
+
+## Additional Considerations
+
+_This is optional. Please add an additional context that would be important to know in order to use this block._
```
Are we requiring documentation here? @matthewroach @JackHowa

matthewroach replied:
This should be cleaned up and at minimum a small description on what the block is - props is not needed

Commit:
Styling error fix

Commit:
updating packages

Commit:
merge with 20

matthewroach committed:
Button should be large based on designs

Commit:
updated readme.md

Commit:
Merge branch 'TBRANDS-29-hero-block' of github.com:WPMedia/arc-themes-blocks into TBRANDS-29-hero-block

nschubach commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,166 @@
+	return (
+		<div className={classes}>
+			<picture>
+				<source srcSet={imageURLDesktop} media={`(${desktopMediaBreakpoint})`} />
```
I think this should be removed and we pick either the 16:9 or the 3:4 image as our MVP.  We can discuss this on the call, but:
* I think it sets a precedent that we don't want to continue (being asked to do this for any image)
* I don't think it's scalable (having to add potentially more breakpoints later)
* I think the conversation should be moved to the "responsive/scalable image" component that we have yet to develop.

I think we should keep this specific use case in mind for the responsive images, but for MVP... I don't think this feature should be implemented at this time.

JackHowa replied:
this has been addressed to some extent. The image could take in an aspect ratio 16:9 or 9:16. But this is still up for work

nschubach commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,166 @@
+			defaultValue: "dark",
+			label: "Content Style",
+		}),
+		imageURLDesktop: PropTypes.string.tag({
+			label: "Image URL for Desktop",
+			searchable: "image",
+		}),
+		desktopMediaBreakpoint: PropTypes.oneOf([
+			"min-width: 800px",
+			"min-width: 700px",
+			"min-width: 600px",
+			"min-width: 500px",
+		]).tag({
+			defaultValue: "min-width: 800px",
+			label:
+				"Image Breakpoint: Enter the media query that will determine when the desktop picture is shown over the mobile version.",
+		}),
+		imageURLMobile: PropTypes.string.tag({
+			label: "Image URL for Mobile",
+			searchable: "image",
+		}),
```
Per my other comment about not doing this right now, we should just have one image and remove the breakpoint logic.

JackHowa replied:
this has been addressed

JackHowa commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,166 @@
+			label: "Image URL for Desktop",
+			searchable: "image",
+		}),
+		desktopMediaBreakpoint: PropTypes.oneOf([
```
per our discussion, this field should be removed in favor of a hard-coded breakpoint. this is assuming that a picture-based implementation would be coming.

JackHowa replied:
this has been addressed

brentmiller1973 replied:
@JackHowa This has been removed.

JackHowa commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,166 @@
+		layout: PropTypes.oneOf(["overlay", "stacked"]).tag({
+			defaultValue: "overlay",
+			label:
+				"Layout configuration. Overlay is text on top of the image and stacked is the text under the image.",
```
Can you add the tooltip descriptions for some of this other description, per our discussions

JackHowa replied:
this has been addressed

JackHowa commented on blocks/hero-block/features/hero/default.jsx:
```diff
@@ -0,0 +1,166 @@
+		}),
+		headline: PropTypes.string,
+		subHeadline: PropTypes.string,
+		description: PropTypes.string,
```
These should have client-facing labels like from ticket: 

    * Headline Text <required field>
    * Sub-Headline Text <optional field>
    * Description Text <optional field>
    * Button #1 Text <required field>
    * Button #1 Link Url <required field>
    * Button #2 Text <optional field>
    * Button #2 Link Url <optional field>

JackHowa replied:
<img width="244" alt="Image URL for Mobile" src="https://user-images.githubusercontent.com/5950956/162067265-6a24d909-55fa-49ba-b0fb-040891232f34.png">

JackHowa replied:
@visionbegin I would expect subHeadline label to be "Sub-Headline Text" per the ticket. don't need a description. I just noticed the lowercase and camelcase naming

brentmiller1973 replied:
How's this?

![image](https://user-images.githubusercontent.com/833461/162235102-43879a99-3f80-4cc9-b11d-d20517395316.png)

brentmiller1973 replied:
The above configuration is pushed.  @JackHowa

JackHowa replied:
thanks !

JackHowa commented on blocks/hero-block/README.md:
```diff
@@ -0,0 +1,23 @@
+# @wpmedia/hero-block
+
+The @wpmedia/hero-block provides a large background visual that is used behind a minimal textual and call-to-action
```
thanks for adding a description

JackHowa replied:
this has been done

JackHowa reviewed (changes requested):
Based on our discussion as a team, I think you listed the takeaways. But I wanted to document the comments as well. When the comments are addressed, can you comment that they are updated. Then I can mark them as resolved

Commit:
Group PR change implementations

JackHowa reviewed (changes requested):
Thanks for making the max-height change per @nschubach. Also, the breakpoint hard-coded for now is pending the image resizer v2. Can you make the last three label changes for headline, sub headline and description proptypes. All we need is a label 

https://github.com/WPMedia/arc-themes-blocks/pull/1292#discussion_r844403588

JackHowa commented on blocks/hero-block/_index.scss:
```diff
@@ -0,0 +1,77 @@
+		@include scss.block-components("hero-stacked");
+	}
+
+	&--light {
```
can you add the dark variant as well? I noticed ``${BLOCK_CLASS_NAME}--dark`` but not corresponding scss class

brentmiller1973 replied:
This has been added @JackHowa

JackHowa replied:
looks good

JackHowa reviewed (changes requested):
--dark in scss per comment should be added

Commit:
Updated prop type lables and descriptions, added dark theme hook

JackHowa reviewed (approved):
Labels look good. approved! thanks for your patience. hopefully, in the future, we can try the all-stakeholder meetings to get qs answered quick

Commit:
Merge branch 'arc-themes-release-version-2.00' into TBRANDS-29-hero-block
